### DISCUSSION TITLE
Edit this page

### DISCUSSION THREAD

**Discussion by zimbatm** [srid/emanote#200] [Ideas]
In order to encourage participation in the content, I would like to have a clear call to action on each footer of the site:

>  This site is open source. Improve this page.

Where "Improve this page." is a link to the repo file on Github.

It's a feature I used to have in Jekyll and has lead to multiple contributions in the past.

**Comment by srid** [comment]
@zimbatm See prior discussion on this at https://github.com/srid/emanote/pull/176 with @edrex

It is a good idea, and can be achieved somehow by overriding `footer.tpl` (see https://github.com/srid/emanote/pull/176#issuecomment-947086353). But it is definitely worth figuring out a simple way to inject edit links to avoid having to override the entire footer. Perhaps we should add two more [hooks](https://github.com/srid/emanote/tree/master/default/templates/hooks) called `footer-{before, after}.tpl`.

**Reply by srid** [reply]
Or you can put the edit link in [`after-note.tpl`](https://github.com/srid/emanote/blob/master/default/templates/hooks/after-note.tpl). We will then need a small change to Emanote to add the relative .md path for using in the template, something like the following (but without the base url). 

https://github.com/srid/emanote/pull/176/files#diff-b2662beb0950ebb033f8848a9866045400f4adb401a9b4a64962bbfd94602788R167-R168

**Reply by srid** [reply]
@zimbatm Okay, I've just implemented this in #201. Add this file to your notebook (ie.,  under `./templates` folder) and it will put the edit link under all notes:

<img width="909" alt="image" src="https://user-images.githubusercontent.com/3998/143774879-fc206811-23d1-442c-8a69-df4b3ad80b35.png">

The diff above you are seeing is for the note.ema.srid.ca site (in `./docs` folder of git repo).

**Reply by zimbatm** [reply]
This works great, thanks! Here is my version :)

```
<pre>
 _____
&lt; <a class="hover:underline text-${theme}-600" href="https://github.com/zimbatm/wiki/edit/master/${ema:note:source-path}">Edit this page on GitHub</a> &gt;
 -----
       \     (\/)
        \   (_o |
             /  |
             \  \______
              \        )o
               /|----- |
               \|     /|
</pre>
```
